Add UI to colony-starter-angular

fuguefoundation opened the pull request:
## Description

Added front end UI using Angular Material.

## Other Changes

Refactored code to use a service to execute Colony code, rather than have it run in the component. This improves modularity.

## Checklist

I didn't have any errors, but would be good to get a second set of eyes on it.

**Added Dependencies**:

[Angular Material](https://material.angular.io)

## Resolves

See [issue 59](https://github.com/JoinColony/colonyStarter/issues/59).

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
-      1,                                // parentDomainId
-    );
+  // Create a colony using the token address of the ERC20 token
+  public async createColony() {
```
The `createColony` method should accept `token` as a parameter. The user should be able to choose what token they would like to use for the native token of their colony.

fuguefoundation replied:
See [commit](https://github.com/JoinColony/colonyStarter/pull/60/commits/4830b5bc16549eacc27e7de37ba0490b636198c2)

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
   }
 
+  // Create a task in the colony
+  public async createTask(title, desc) {
```
The `createTask` method should accept `domainId` as a parameter. The user should be able to choose the domain for the task they are creating.

fuguefoundation replied:
See [commit](https://github.com/JoinColony/colonyStarter/pull/60/commits/4830b5bc16549eacc27e7de37ba0490b636198c2)

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
+    this.model.task.title = title;
+    this.model.task.desc = desc;
+    this.appService.createTask(this.state.colonyClient[0], this.model.domain.id, taskObj).then( (res) => {
+      console.log(res);
```
Remove console.log. Maybe you forgot to remove after testing.

fuguefoundation replied:
Removed all console logs

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
+  // Add a domain to the colony
+  public async addDomain() {
+    this.appService.addDomain(this.state.colonyClient[0], this.model.parentDomainId).then( (res) => {
+      console.log(res);
```
Remove `console.log`. Maybe you forgot to remove after testing.

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
+  // Get an initialized ColonyClient for the colony
+  public async getColonyClient() {
+    this.appService.getColonyClient(this.state.networkClient[0], this.model.colony.id).then( (res) => {
+      console.log(res);
```
Remove console.log. Maybe you forgot to remove after testing.

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
+  // Create a colony using the token address of the ERC20 token
+  public async createColony() {
+    this.appService.createColony(this.state.networkClient[0], this.model.token.address).then( (res) => {
+      console.log(res);
```
Remove console.log. Maybe you forgot to remove after testing.

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
-    );
+    public async token(name, symbol) {
+    this.appService.createToken(this.state.networkClient[0], name, symbol).then( (res) => {
+      console.log(res);
```
Remove console.log. Maybe you forgot to remove after testing.

ryanchristo commented on packages/colony-starter-angular/src/app/app.component.ts:
```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
     // Create an instance of ColonyNetworkClient using the adapter
-    this.state.networkClient[0] = await connectNetwork(0);
+    this.appService.connectNetwork().then( (res) => {
+      console.log(res);
```
Remove console.log. Maybe you forgot to remove after testing.

fuguefoundation replied:
Removed

fuguefoundation commented:
Hey @ryanchristo, thanks for your fast feedback, very helpful to seeing what right looks like when collaborating on an open source project!

I'll take a look at these edits/issues in the coming days, but a few follow up questions/comments.

1. Most for my own curiosity, what is the significance of a capital letter for commit messages?
2. I put 'first commit' for the new `app.service.ts` file but I can see how even if it's a new file that's not super helpful
3. I added in node-metamask because, well because it's Metamask and this feature is awesome, but I kept getting errors related to `fs` dependency so I removed it. This is something I've encountered in my own project and haven't had a chance to figure out how to get it working inside Angular.
4. Everything else looks fine and doable, thanks!

ryanchristo commented:
@fuguefoundation Awesome! Looking forward to the revisions.

(1) We like to keep things consistent across our entire codebase.
(2) Something more descriptive would be much appreciated here, thanks!
(3) The `node-metamask` package might not be what you are looking for since you are building a boilerplate for a dApp that will be able to interact with MetaMask via the browser. I would hold off on adding anything MetaMask related here since we are creating a boilerplate that is focused on getting a local development environment set up for developers, but you should check out [purser](https://github.com/JoinColony/purser) when you get a chance, which might be more along the lines of what you were looking for.
(4) Awesome, let me know if you have any questions.

fuguefoundation commented:
Nice, Purser looks awesome. Trezor + Metamask is exactly what I'm looking for.

fuguefoundation commented:
@ryanchristo - Updated with recent commits. I addressed everything mentioned above except breaking out the code into separate, smaller components. I looked at `colony-starter-react` (which has more functionality than this project) and agree that modularity is a good design pattern, but maybe that is something to visit further down the road. `app.service.ts`, which actually invokes the Colony example js files, does help with modularity to an extent. Anyway, let me know your thoughts on the new features and anything else.
